react/week4/src/components/UserList.js

```diff
@@ -0,0 +1,32 @@
+import React, { useContext } from "react";
+import { userListContext } from "../App";
+import EmptyResult from "./EmptyResult";
+
+const UserList = () => {
+  const { user } = useContext(userListContext);
+  return (
+    <ul>
+      {user.total_count === undefined ? (
+        <EmptyResult />
+      ) : (
+        user.items.map(list => {
+          return (
+            <>

```

Comment:
I would use this one as a list item, considering we have a `ul` element. Also, to this one, we need to add a `key` attribute, otherwise React will complain `Each child in a list should have a unique "key" prop.` and then change the `<li>` element you already have to a `div`. The reason we need to do this is that your entire segment `<>` is a list item, not just the `<li key={list.id}>{list.login}</li>` element.
```suggestion
            <li key={list.id}>
```

Reply:
corrected as  <li key={user.id}>

Reply:
corrected as ` <li key={list.id}>`